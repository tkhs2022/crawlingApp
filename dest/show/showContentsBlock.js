"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.MoveBlockOnClick = exports.ContentsBlockControl = void 0;

var _react = _interopRequireDefault(require("react"));

var _cardComponent = _interopRequireDefault(require("../card/cardComponent.jsx"));

var _showContentsArea = require("./showContentsArea.jsx");

var _store = _interopRequireDefault(require("../store/store.js"));

var _commonFunc = require("../commonFunc.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } else if (call !== void 0) { throw new TypeError("Derived constructors may only return object or undefined"); } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

///////////////////////////////////////////////////////////////// 
// コンテンツブロックをコントロール。コンテンツごとに記事を振り分け。(初回のみ)
var ContentsBlockControl = /*#__PURE__*/function (_React$Component) {
  _inherits(ContentsBlockControl, _React$Component);

  var _super = _createSuper(ContentsBlockControl);

  function ContentsBlockControl(props) {
    _classCallCheck(this, ContentsBlockControl);

    return _super.call(this, props);
  }

  _createClass(ContentsBlockControl, [{
    key: "componentWillMount",
    value: function componentWillMount() {
      // storeアクセスでの待機中はローディングCSSを流す
      (0, _commonFunc.cssFileAble)("loading.css");
    }
  }, {
    key: "componentDidMount",
    value: function componentDidMount() {
      (0, _commonFunc.cssFileDisable)("loading.css");
    }
  }, {
    key: "render",
    value: function render() {
      var _this = this;

      var contentesList = _store.default.getState().componentReducer.thisContents;

      var Rows = contentesList != undefined && contentesList.article.map(function (Content, index) {
        return _this.props.Kbn == Content.kbn && /*#__PURE__*/_react.default.createElement(_cardComponent.default, {
          key: index,
          Content: Content
        });
      });
      return (// デフォルトでは区分1のコンテンツブロックを表示
        this.props.Kbn == 1 ? /*#__PURE__*/_react.default.createElement("div", {
          id: this.props.Kbn,
          className: _showContentsArea.DetailShow
        }, Rows) : /*#__PURE__*/_react.default.createElement("div", {
          id: this.props.Kbn,
          className: _showContentsArea.DetailOriginal
        }, Rows)
      );
    }
  }]);

  return ContentsBlockControl;
}(_react.default.Component); ///////////////////////////////////////////////////////////////// 
// コンテンツボックスのアニメーション表示・非表示。


exports.ContentsBlockControl = ContentsBlockControl;

var MoveBlockOnClick = function MoveBlockOnClick(TargetKbn, Caller) {
  var SiteHedderActive = _showContentsArea.SiteHedderActiveM.GetterSiteHedderActive();

  if (TargetKbn == null && TargetKbn == undefined && Caller == _showContentsArea.CaseA) {
    return;
  } // ボタンクリックのあったコンテンツタイトルに対応するコンテンツブロックを取得


  var ElementOfContentsBlock = document.getElementById(TargetKbn); // コンテンツブロックの表示状態(className)を取得

  var TargetContentsBlockClassName = ElementOfContentsBlock.getAttribute("class");

  switch (Caller) {
    // CaseA:履歴表示ボタン処理から呼ばれたケース
    case _showContentsArea.CaseA:
      switch (TargetContentsBlockClassName) {
        case _showContentsArea.DetailShow:
          SiteHedderActive && ElementOfContentsBlock.setAttribute("class", _showContentsArea.DetailShowActiveSiteHedder);
          break;

        case _showContentsArea.DetailShowActiveSiteHedder:
          !SiteHedderActive && ElementOfContentsBlock.setAttribute("class", _showContentsArea.DetailShow);
          break;
      }

      break;
    // CaseB:コンテンツタイトルボタン押下処理から呼ばれたケース

    case _showContentsArea.CaseB:
      switch (TargetContentsBlockClassName) {
        case _showContentsArea.DetailOriginal:
          SiteHedderActive ? ElementOfContentsBlock.setAttribute("class", _showContentsArea.DetailShowActiveSiteHedder) : ElementOfContentsBlock.setAttribute("class", _showContentsArea.DetailShow);
          break;

        case _showContentsArea.DetailHide:
          SiteHedderActive ? ElementOfContentsBlock.setAttribute("class", _showContentsArea.DetailShowActiveSiteHedder) : ElementOfContentsBlock.setAttribute("class", _showContentsArea.DetailShow);
          break;

        case _showContentsArea.DetailShow:
          ElementOfContentsBlock.setAttribute("class", _showContentsArea.DetailHide);
          break;

        case _showContentsArea.DetailShowActiveSiteHedder:
          ElementOfContentsBlock.setAttribute("class", _showContentsArea.DetailHide);
          break;
      }

  }
};

exports.MoveBlockOnClick = MoveBlockOnClick;
var _default = MoveBlockOnClick;
exports.default = _default;