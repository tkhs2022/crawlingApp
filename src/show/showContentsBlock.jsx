
///////////////////////////////////////////////////////////////// 
// コンテンツボックスに記事を表示するクラス
///////////////////////////////////////////////////////////////// 
import React from "react";
import MediaCardForContentsBlock from '../card/cardComponent.jsx';
// コンテンツボックス部分のスタイル変数
import {DetailOriginal, DetailShow, DetailHide, DetailShowActiveSiteHedder} from './showContentsArea.jsx';
import {SiteHedderActiveM} from './showContentsArea.jsx';
// MoveBlockOnClick関数呼び出し時のパラメータに、自身の関数名を入れる。
import {CaseA, CaseB} from './showContentsArea.jsx';
import store from "../store/store.js";
import { cssFileAble, cssFileDisable } from '../commonFunc.js';
///////////////////////////////////////////////////////////////// 
// コンテンツブロックをコントロール。コンテンツごとに記事を振り分け。(初回のみ)
export class ContentsBlockControl extends React.Component {
  constructor(props) {
    super(props);
  }

  componentWillMount() {
    // storeアクセスでの待機中はローディングCSSを流す
    cssFileAble("loading.css");
  }

  componentDidMount() {
    cssFileDisable("loading.css");
  }

  render() {
    var contentesList = store.getState().componentReducer.thisContents;
    const Rows = 
     contentesList != undefined &&
      contentesList.article.map((Content, index) => {
        return(
          this.props.Kbn == Content.kbn &&
            <MediaCardForContentsBlock key={index} Content={Content}/>
        );
      });
    
    return(
      // デフォルトでは区分1のコンテンツブロックを表示
      this.props.Kbn == 1 ?
        <div id ={this.props.Kbn} className={DetailShow}>{Rows}</div> :
        <div id ={this.props.Kbn} className={DetailOriginal}>{Rows}</div>
      );
  }
}
   
///////////////////////////////////////////////////////////////// 
// コンテンツボックスのアニメーション表示・非表示。
export const MoveBlockOnClick = (TargetKbn, Caller) => {
  var SiteHedderActive = SiteHedderActiveM.GetterSiteHedderActive();
  
  if (TargetKbn == null && TargetKbn == undefined && Caller == CaseA) {
    return;
  }

  // ボタンクリックのあったコンテンツタイトルに対応するコンテンツブロックを取得
  var ElementOfContentsBlock = document.getElementById(TargetKbn);
  // コンテンツブロックの表示状態(className)を取得
  var TargetContentsBlockClassName = ElementOfContentsBlock.getAttribute("class");

  switch (Caller) {
    // CaseA:履歴表示ボタン処理から呼ばれたケース
    case CaseA:
      switch (TargetContentsBlockClassName) {
        case DetailShow:
          SiteHedderActive && ElementOfContentsBlock.setAttribute("class",DetailShowActiveSiteHedder);
          break;
        case DetailShowActiveSiteHedder:
          !SiteHedderActive && ElementOfContentsBlock.setAttribute("class",DetailShow);
          break;
      }
      break;

    // CaseB:コンテンツタイトルボタン押下処理から呼ばれたケース
    case CaseB:
      switch (TargetContentsBlockClassName) {
        case DetailOriginal:
          SiteHedderActive ? ElementOfContentsBlock.setAttribute("class",DetailShowActiveSiteHedder) : ElementOfContentsBlock.setAttribute("class",DetailShow);         
          break;
        case DetailHide:
          SiteHedderActive ? ElementOfContentsBlock.setAttribute("class",DetailShowActiveSiteHedder) : ElementOfContentsBlock.setAttribute("class",DetailShow);
          break;
        case DetailShow:
          ElementOfContentsBlock.setAttribute("class",DetailHide);
          break;
        case DetailShowActiveSiteHedder:
          ElementOfContentsBlock.setAttribute("class",DetailHide);
          break;
      }
  }
};

export default MoveBlockOnClick;