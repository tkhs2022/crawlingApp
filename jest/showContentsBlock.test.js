import React from 'react';
import renderer from 'react-test-renderer';
import ContentsList from '../src/data/ContentsList.js';
import { ContentsBlockControl } from '../src/show/showContentsBlock.jsx';

const Contents = new ContentsList(ContentsList);
const thisContents = Contents.getContentsList("contents.json");

test('showContentsBlock.jsx', () => {
	const index = 2;
	const kbn = 2;
	const component = renderer.create(
		<ContentsBlockControl key={index} Kbn ={kbn} Contents={thisContents}/>,
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});