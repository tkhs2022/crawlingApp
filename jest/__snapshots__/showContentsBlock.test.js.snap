// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`showContentsBlock.jsx 1`] = `
<div
  className="detail"
  id={2}
>
  <div
    className="article"
  >
    <div
      className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        className="MuiButtonBase-root MuiCardActionArea-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <a
          href="https://jbpress.ismedia.jp/articles/-/66896"
          rel="noreferrer"
          style={
            Object {
              "textDecoration": "none",
            }
          }
          target="_blank"
        >
          <div
            className="MuiCardMedia-root makeStyles-media-2"
            style={
              Object {
                "backgroundImage": "url(\\"https://jbpress.ismcdn.jp/mwimgs/a/2/600mw/img_a2b3bf48688c17f97cc0b70824e4359e502174.jpg\\")",
              }
            }
            title="日本人に「石の上にも三年」よりも「逃げ恥」が重要と思う理由"
          />
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              JBPress
            </h2>
            <h2
              className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
            >
              日本人に「石の上にも三年」よりも「逃げ恥」が重要と思う理由
            </h2>
          </div>
        </a>
        <span
          className="MuiCardActionArea-focusHighlight"
        />
      </button>
    </div>
  </div>
  <div
    className="article"
  >
    <div
      className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        className="MuiButtonBase-root MuiCardActionArea-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <a
          href="https://jbpress.ismedia.jp/articles/-/66719"
          rel="noreferrer"
          style={
            Object {
              "textDecoration": "none",
            }
          }
          target="_blank"
        >
          <div
            className="MuiCardMedia-root makeStyles-media-2"
            style={
              Object {
                "backgroundImage": "url(\\"https://jbpress.ismcdn.jp/mwimgs/2/5/600mw/img_250c1e03181a397a1fa68c3a76e7b91476218.jpg\\")",
              }
            }
            title="8.29写真ルポ：灯が消えた池袋マルイ「最後の日」"
          />
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              JBPress
            </h2>
            <h2
              className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
            >
              8.29写真ルポ：灯が消えた池袋マルイ「最後の日」
            </h2>
          </div>
        </a>
        <span
          className="MuiCardActionArea-focusHighlight"
        />
      </button>
    </div>
  </div>
</div>
`;
